### PR TITLE
Remove Argo CD integration, and update Argo CD dependencies

### PR DESCRIPTION
**Pull request by jgwest**
This PR:
- Updates the Go version to 1.17
- Updates Go module dependencies, and Dockerfile base image, to same versions as Argo CD 
- Removes the integrated ApplicationSet+Argo CD install YAML (this is now handled at the Argo CD level)
    - Updates docs
    - Removes generated YAML, and kustomize
    - Updates corresponding build/release shell scripts
- Regenerates ApplicationSet/Argo CD manifests, to latest from 2.3.0-rc5

### CODE REVIEW

File: docs/Getting-Started.md [hunk 1]
```diff
@@ -34,19 +43,6 @@ The `manifests/install.yaml` file contains the Kubernetes manifests required to
 - RoleBinding to bind the ServiceAccount and Role
 
 
-### B) Install ApplicationSet and Argo CD together
```
Comment (crenshaw-dev) [inline docs/Getting-Started.md:37]: `C` below should be changed to `B`.

File: go.mod [hunk 2]
```diff
@@ -15,48 +15,177 @@ require (
...
+	k8s.io/api v0.23.1
+	k8s.io/apiextensions-apiserver v0.23.1
+	k8s.io/apimachinery v0.23.1
+	k8s.io/client-go v0.23.1
+	k8s.io/utils v0.0.0-20210930125809-cb0fa318a74b
+	sigs.k8s.io/controller-runtime v0.11.0
+	sigs.k8s.io/yaml v1.3.0
+)
+
+require (
```
Comment (crenshaw-dev) [inline go.mod:29]: Just for my own knowledge: is this just a pinning mechanism for transient dependencies, in order to get (relatively) deterministic builds?

File: hack/generate-manifests.sh [hunk 3]
```diff
@@ -42,11 +42,5 @@ cd ${SRCROOT}/manifests/namespace-install && ${KUSTOMIZE} build . >> ${TEMPFILE}
 mv ${TEMPFILE} ${SRCROOT}/manifests/install.yaml
 cd ${SRCROOT} && chmod 644 manifests/install.yaml
 
-# Verify that the 'install-with-argo-cd.yaml' is targetting the expected Argo CD version
+# Verify that the GitHub actions is targetting the expected Argo CD version
```
Comment (crenshaw-dev) [inline hack/generate-manifests.sh:45]:
```suggestion
# Verify that the GitHub actions are targeting the expected Argo CD version
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: hack/generate-manifests.sh (hunk 3 above)
Comment (wtam2018) [inline hack/generate-manifests.sh:45]:
```suggestion
# Verify that the GitHub Actions is targeting the expected Argo CD version
```

**Review by wtam2018 — APPROVED** [review]
LGTM
thanks @jgwest !

**Pull request merged**